framework/include/base/MooseObjectWarehouseBase.h

```diff
@@ -242,6 +244,25 @@ MooseObjectWarehouseBase<T>::addObject(MooseSharedPointer<T> object, THREAD_ID t
 
 
 template<typename T>
+inline MooseSharedPointer<T>
+MooseObjectWarehouseBase<T>::getObject(const std::string & name, THREAD_ID tid/* = 0*/) const
+{
+  checkThreadID(tid);
+  MooseSharedPointer<T> output;
+  typename std::vector<MooseSharedPointer<T> >::const_iterator it;
+  for (it = _all_objects[tid].begin(); it != _all_objects[tid].end(); ++it)
+  {
+    if ((*it)->name() == name)
+    {
+      output = *it;
+      break;
+    }
+  }
+  return output;
```

Comment:
Looks like it's possible to return an empty pointer. If that's the design you should probably handle that in the callers below, else you should throw an error here.